src/protosym/core/evaluate.py

```diff
@@ -26,30 +26,91 @@
 
 
 if _TYPE_CHECKING:
-    from typing import Optional, Iterable
+    from typing import Optional, Iterable, Any
 
     Op1 = Callable[[_T], _T]
     Op2 = Callable[[_T, _T], _T]
     OpN = Callable[[Iterable[_T]], _T]
 
 
+def _generic_operation_error(head: TreeExpr, argvals: Iterable[_T]) -> _T:
+    """Error fallback rule for handling unknown heads."""
+    msg = "No rule for head: " + repr(head)
+    raise NoEvaluationRuleError(msg)
+
+
+def _generic_atom_error(value: TreeAtom[_S]) -> Any:
```

Comment:
Is this `Any` type hint for the return correct if an error is being raised before implicit `None` would be returned?

Reply:
Same as the other case.

The reason this is typed as `Any` rather than `_T` is because you can't return `_T` if there is no `_T` in the parameters.

Reply:
Done